### PR TITLE
PMM-5809 sandbox fixes

### PR DESCRIPTION
**Pull request by percona-csalguero**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -10,6 +10,9 @@ jobs:
   build:
     name: Build
     runs-on: ubuntu-latest
+    strategy:
+        matrix:
+            version: [3.6, 4.0, 4.2 ]
```
Comment (AlekSi) [inline .github/workflows/go.yml]: Extra space before `]` or missing space before `[`

File: docker-compose.yml [hunk 2]
```diff
@@ -2,31 +2,31 @@ version: '3'
 services:
     mongo-1-1:
         container_name: "mongo-1-1"
-        image: mongo:4.0.1
+        image: ${MONGODB_FLAVOR:-mongo}:${VERSION:-4.0.1}
```
Comment (AlekSi) [inline docker-compose.yml]: Why that version specifically by default?

File: .github/workflows/go.yml [hunk 3]
```diff
@@ -10,6 +10,9 @@ jobs:
   build:
     name: Build
     runs-on: ubuntu-latest
+    strategy:
+        matrix:
```
Comment (AlekSi) [inline .github/workflows/go.yml:14]: Should we also add `MONGODB_FLAVOR` to the matrix to test with PSMDB?

Comment (percona-csalguero) [reply .github/workflows/go.yml:14]: We are not using any special Percona implemented feature so, right now, there is no need to test with Percona Server for Mongodb and MongoDB itself.

File: docker-compose.yml [hunk 4]
```diff
@@ -2,31 +2,31 @@ version: '3'
...
         ports:
-            - "${TEST_MONGODB_S1_SECONDARY2_PORT}:27017"
-        command: mongod --replSet rs1 --shardsvr --port 27017  --oplogSize 16 --noprealloc --smallfiles
+            - "${TEST_MONGODB_S1_SECONDARY2_PORT:-17003}:27017"
+        command: mongod --replSet rs1 --shardsvr --port 27017  --oplogSize 16 
 
     mongo-rs1-setup:
         container_name: "mongo-rs1-setup"
-        image: mongo:4.0.1
+        image: ${MONGODB_FLAVOR:-mongo}:${VERSION:-4.0.1}
```
Comment (BupycHuk) [inline docker-compose.yml]: Why don't use one env variable for the image instead of two?

File: .github/workflows/go.yml (hunk 3 above)
Comment (BupycHuk) [reply .github/workflows/go.yml:14]: I think we should test it anyway.

File: docker-compose.yml (hunk 2 above)
Comment (percona-csalguero) [reply docker-compose.yml]: no reason. inherited from old sandbox test

File: .github/workflows/go.yml [hunk 5]
```diff
@@ -10,6 +10,9 @@ jobs:
...
+            image: [ "mongo:3.6", "mongo:4.0", "mongo:4.2", "percona/percona-server-mongodb:3.6", "percona/percona-server-mongodb:4.0" ]
```
Comment (BupycHuk) [inline .github/workflows/go.yml]: I would suggest to keep each version in separate line.

**Review by BupycHuk — APPROVED** [review]
LGTM

File: .github/workflows/go.yml (hunk 3 above)
Comment (AlekSi) [reply .github/workflows/go.yml:14]: Now we have `TEST_MONGODB_FLAVOR` defined in Makefile that is not used?

File: .github/workflows/go.yml (hunk 5 above)
Comment (AlekSi) [reply .github/workflows/go.yml]: This ^

File: .github/workflows/go.yml [hunk 6]
```diff
@@ -27,8 +30,9 @@ jobs:
 
     - name: Test
       run: |
-        make test-cluster
+        MONGODB_IMAGE=${{ matrix.image }} make test-cluster
```
Comment (AlekSi) [inline .github/workflows/go.yml]: It is `MONGODB_VERSION` below.

**Review by AlekSi — CHANGES_REQUESTED** [review]

**Comment by percona-csalguero** [comment]
@AlekSi I removed Percona Server from the matrix and created https://jira.percona.com/browse/PMM-6350 because config servers are failing with PSMDB

**Review by BupycHuk — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]

**Pull request merged**